pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -577,11 +577,106 @@ func (r *Reconciler) processTargetPools(desired bool, poolFunc poolProcessor) er
 	return nil
 }
 
+func (r *Reconciler) ensureInstanceGroup(instanceGroupName string) error {
+	if err := r.checkInstanceGroupExists(instanceGroupName); err != nil {
```

Comment:
Does this error always mean we need to create a new instance group? Can we check the error for a 404 somehow please